### PR TITLE
Rest update to use cucumber, verification steps moved to utils

### PR DESCRIPTION
**Pull request by tplevko**

### CODE REVIEW

File: rest-tests/src/test/resources/features/common.feature [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
+  @integrations-twitter-to-salesforce
+  Scenario: Tw - sf integration
+    Given user cleans tw to sf scenario
+    And user creates the tw connection using "twitter_listen" template
+    And user creates sf connection
+    And user creates tw mention step with "twitter-mention-connector" action
+    And user creates basic tw to sf filter step
+    And user creates tw to sf mapper step
+    And user creates sf step for tw sf test
+    And user creates tw to sf integration
```
Comment (dsimansk) [inline rest-tests/src/test/resources/features/common.feature]: I'd suggest to change this a bit a split this to 
```
When user creates tw to sf integration
Then wait for integration "<IntegrationName>" to be "Active"
```
wdyt?

File: rest-tests/src/test/resources/features/common.feature [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
+  Scenario: Clean & deploy
+    Given clean default namespace
+    When deploy Syndesis from template
+    Then wait for Syndesis to become ready
+
+  @integrations-twitter-to-salesforce
+  Scenario: Tw - sf integration
+    Given user cleans tw to sf scenario
+    And user creates the tw connection using "twitter_listen" template
+    And user creates sf connection
```
Comment (dsimansk) [inline rest-tests/src/test/resources/features/common.feature]: I don't see this step implementation, is it possible? :)

File: rest-tests/src/test/resources/features/common.feature [hunk 3]
```diff
@@ -1,7 +1,32 @@
 Feature: Common scenarios
 
-@clean-deploy
-Scenario: Clean & deploy
-	Given clean default namespace
-	When deploy Syndesis from template
-	Then wait for Syndesis to become ready
\ No newline at end of file
+  @clean-deploy
```
Comment (dsimansk) [inline rest-tests/src/test/resources/features/common.feature]: Lets remove clean-deploy for now and stick just with `reset-db`

File: rest-tests/src/test/java/io/syndesis/qe/rest/tests/integrations/TwSfSteps.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	private final List<Step> steps = new ArrayList<>();
+	private AccountsDirectory accountsDirectory;
+
+	public TwSfSteps() throws GeneralSecurityException {
+		accountsDirectory = new AccountsDirectory();
+		connectorsEndpoint = new ConnectorsEndpoint(RestConstants.getInstance().getSyndesisURL());
+		connectionsEndpoint = new ConnectionsEndpoint(RestConstants.getInstance().getSyndesisURL());
+	}
+
+	@Given("^user creates tw mention step with \"([^\"]*)\" action")
```
Comment (dsimansk) [inline rest-tests/src/test/java/io/syndesis/qe/rest/tests/integrations/TwSfSteps.java]: Maybe we could have one generic method  like`user creates step from connector \"([^\"]*)\" with \"([^\"]*)\" action`

File: rest-tests/src/test/resources/features/common.feature (hunk 2 above)
Comment (tplevko) [reply rest-tests/src/test/resources/features/common.feature]: all of the steps are implemented somewhere, it builds with no problem :)

File: rest-tests/src/test/resources/features/common.feature (hunk 1 above)
Comment (tplevko) [reply rest-tests/src/test/resources/features/common.feature]: ok, sounds reasonable

File: rest-tests/src/test/resources/features/common.feature (hunk 2 above)
Comment (tplevko) [reply rest-tests/src/test/resources/features/common.feature]: added, srry my bad

**Comment by tplevko** [comment]
Updated. The generic steps you mentioned would be quite problematic to implement, there are many moving parts there, I will think about possible solution, but for now I think it is for the best to leave it as it is. The other things you highlighted should be reflected.

**Pull request merged**